### ISSUE TITLE
Eslint lint API not working

### ISSUE THREAD

**Issue by beckend** [typescript-eslint/typescript-eslint#4210]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/linting/TROUBLESHOOTING.md) and my problem is not listed.

**Repro**

.eslintrc
```JSON
{
  "parser": "@typescript-eslint/parser",
  "plugins": ["@typescript-eslint"]
}
```

```TS
import { join } from 'path'
import { ESLint } from 'eslint'

const paths = {
  dir: {
    fixtures: join(__dirname, 'fixtures'),
    root: join(__dirname, '../../..'),
  },
}

describe('linter', () => {
  describe('errors', () => {
    it('when errors are present', async () => {
      const eslint = new ESLint()
      const results = await eslint.lintFiles([
        join(paths.dir.fixtures, 'error/1.ts.fixture'),
      ])

      expect(results.length).toBeGreaterThan(0)

      results.forEach(({ errorCount }) => {
        expect(errorCount).toBeGreaterThan(0)
      })
    })
  })
})
```

**Expected Result**
To lint properly

**Actual Result**
```shell
Failed to load plugin '@typescript-eslint' declared in '.eslintrc.js': Cannot find module 'eslint/use-at-your-own-risk' from 'node_modules/@typescript-eslint/eslint-plugin/dist/util/getESLintCoreRule.js'

    Require stack:
      node_modules/@typescript-eslint/eslint-plugin/dist/rules/brace-style.js
      node_modules/@typescript-eslint/eslint-plugin/dist/rules/index.js
      node_modules/@typescript-eslint/eslint-plugin/dist/index.js
      node_modules/@eslint/eslintrc/dist/eslintrc.cjs
      node_modules/eslint/lib/cli-engine/cli-engine.js
      node_modules/eslint/lib/eslint/eslint.js
      node_modules/eslint/lib/eslint/index.js
      node_modules/eslint/lib/api.js
      src/eslint/__tests__/base.spec.ts

    Referenced from: /home/admin/path/to/.eslintrc.js

      at Resolver.resolveModule (node_modules/jest-resolve/build/resolver.js:324:11)
      at node_modules/@typescript-eslint/eslint-plugin/src/util/getESLintCoreRule.ts:47:11
      at Object.<anonymous> (node_modules/@typescript-eslint/eslint-plugin/src/rules/brace-style.ts:10:35)

```

**Additional Info**
Recursive listing files and folder in `node_modules/eslint` and finding `use-at-your-own-risk'` was not a hit.
The code in question: https://github.com/typescript-eslint/typescript-eslint/blob/6c3816b3831e6e683c1a7842196b34248803d69b/packages/eslint-plugin/src/util/getESLintCoreRule.ts#L47

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.4.0` |
| `@typescript-eslint/parser`        | `5.4.0` |
| `TypeScript`                       | `4.5.2` |
| `ESLint`                           | `8.3.0` |
| `node`                             | `16.13.0` |

**Comment by armano2** [comment]
info: as for `use-at-your-own-risk` you can find it here:

https://github.com/eslint/eslint/blob/e03a96755dcda1caa7b6bb21f0f6db843d0cf799/package.json#L10-L14
```
  "exports": {
    "./package.json": "./package.json",
    ".": "./lib/api.js",
    "./use-at-your-own-risk": "./lib/unsupported-api.js"
  },
```

i'm not sure what is a root issue here, but it looks that there node has an issue with correctly resolving this file :(

**Comment by bradzacher** [comment]
If I had to guess - you have multiple eslint versions installed and this is causing some weirdness.
That's usually the cause of such issues.

Esp if your lint run is working normally.

**Comment by icatalina** [comment]
Same issue, it only happens when ran through jest though, if I run the same file directly with node it works correctly.

**Comment by icatalina** [comment]
`use-at-your-own-risk` is present in eslint's `package.json` as an alias for `./lib/unsupported-api.js`, which does exist.

https://github.com/eslint/eslint/blob/main/package.json#L13

**Comment by bradzacher** [comment]
Oh your comment just reminded me.

Jest currently does not support ESM out of the box.
This is beyond our control.

https://github.com/facebook/jest/issues/9771#issuecomment-871585234

https://github.com/typescript-eslint/typescript-eslint/blob/eaaa2047ca54f098dcdd32aaf5d8949495c6be26/tests/jest-resolver.js

**Comment by icatalina** [comment]
https://github.com/bigcommerce/eslint-config/pull/108

**Comment by ktalebian** [comment]
What's the solution? I'm getting this same error and I am on ESLint 8.4 and latest version of this module.

**Comment by bradzacher** [comment]
read my comment. it includes links to workarounds.
